test/image-blueprints/layer5-bootc/group1/cos9-bootc-source.containerfile

```diff
@@ -0,0 +1,59 @@
+FROM quay.io/centos-bootc/centos-bootc:stream9
+
+# Build arguments
+ARG PREVIOUS_MINOR_VERSION=15
+ARG USHIFT_RPM_REPO_NAME=microshift-local
+ARG USHIFT_RPM_REPO_PATH=/tmp/$USHIFT_RPM_REPO_NAME
+
+# Copy the MicroShift repository contents
+ADD $USHIFT_RPM_REPO_NAME $USHIFT_RPM_REPO_PATH
+
+# MicroShift local RPM repository
+RUN printf "\
+[microshift-local]\n\
+name=MicroShift Local Repository\n\
+baseurl=$USHIFT_RPM_REPO_PATH\n\
+enabled=1\n\
+gpgcheck=0\n\
+skip_if_unavailable=0\n" > "/etc/yum.repos.d/microshift-local.repo"
+
+# OpenShift Mirror beta RPM repository for MicroShift dependencies
+# Only the released previous minor version is guaranteed to be available
+RUN printf "\
+[openshift-mirror-beta]\n\
+name=OpenShift Mirror Beta Repository\n\
+baseurl=https://mirror.openshift.com/pub/openshift-v4/$(uname -m)/dependencies/rpms/4.${PREVIOUS_MINOR_VERSION}-el9-beta/\n\
+enabled=1\n\
+gpgcheck=0\n\
+skip_if_unavailable=0\n" > "/etc/yum.repos.d/openshift-mirror-beta.repo"
+
+# Implement workarounds necessary for the successful MicroShift operation
+# - OVN-k copies an executable to /opt/cni/bin directory on the host, which is
+#   read-only on bootc images. Create the /opt/cni/bin -> /var/opt/cni/bin
+#   symbolic link as a workaround.
+# - The /var/run directory must be a symbolic link to /run, which is not
+#   the case in bootc images. This cases problems with services like DBus
+#   and NetworkManager. DBus creates its socket in /run/dbus directory, which
+#   is accessed by NetworkManager at /var/run/dbus.
+RUN mkdir -p /opt/cni && mkdir -p /var/opt/cni/bin && ln -s /var/opt/cni/bin /opt/cni/ ; \
+    [ ! -L /var/run ] && rm -rf /var/run && ln -s /run /var/
+
+# Install MicroShift, few helper utilities and cleanup
+RUN dnf install -y vi firewalld microshift && \
```

Comment:
If we separate the helpers from microshift, we'll be able to reuse more of the image layers in CI. We should probably set up the layers as

- OVN-K directory massaging pre-reqs
- crio configuration
- firewall installation
- rpm repo configuration
- microshift installation

The last 2 layers are going to change the most often across the images we build for the various test scenarios.

Reply:
Yes, once we have more configurations in place, we will have an opportunity to create layering. 
At this time, we only have one configuration, so I'd postpone layering for later when we have more combinations.